fixes #7887 - rpm scoped search conversion

jlsherrill opened the pull request:
TODO: 

- [x] Ruby model tests
- [x] Ruby controller tests
- [x] javascript tests
- [x] test/double check content view filter logic

jlsherrill commented:
TODO: 

- [ ] Ruby tests
- [ ] javascript tests
- [ ] test/double check content view filter logic

ehelms commented:
I think the remove packages from a repository logic needs updating from glancing at this (was hoping to copy your implementation for puppet modules :)

jlsherrill commented:
a handful of new tests to write but ready for review

ehelms commented:
@jlsherrill mind rebasing this and I'll start testing?

jlsherrill commented:
@ehelms done

jlsherrill commented:
[test]

ehelms commented on lib/katello/tasks/upgrades/2.4/import_rpms.rake:
```diff
@@ -0,0 +1,12 @@
+namespace :katello do
+  namespace :upgrades do
+    namespace '2.1' do
```
2.4

ehelms replied:
Also need to include this in the engine so it gets loaded and open a PR to the installer to include this

ehelms commented:
1) Strangely it isn't suggesting x86_64 for autocomplete when doing arch.
2) I get no autocomplete suggestions for checksum
3) *.el7 changes itself to  0.0.el7 -- can I search by 'dist' in the release?
4) Not specifically this PRs issue, but if you view a RHEL repo, and click the package link it takes you to the list of packages with the remove action available which throws a 500 when you try but never shows the user an error.
5) Should the 'Packages' menu link be 'RPMs' ?

ehelms commented:
6) Select checkboxes exist next to RPMs on the 'Packages' page but no actions are present
7) No summaries are present for any of the RPMs ?
8) No packages seem to  show anything under 'Requires' when browsing the dependencies tab.
9) When viewing an errata, it lists the affected packages, what about turning those into links to the package details pages for each (could be a future issue) ?

ehelms commented on app/controllers/katello/api/v2/api_controller.rb:
```diff
@@ -80,10 +80,15 @@ def scoped_search(query, default_sort_by, default_sort_order, options = {})
+      if sort_attr
+        sort_attr = "#{query.table_name}.#{sort_attr}" unless sort_attr.to_s.include?('.')
+        query = query.order("#{sort_attr} #{params[:sort_order] || default_sort_order}")
+      elsif options[:custom_sort]
```
Maybe this should be checked for first since if you are providing this option, the intent would be for it to be processed , otherwise `sort_attr` should be processed?

jlsherrill replied:
well if the user passes in sort attributes we want to use those first

ehelms commented on app/controllers/katello/api/v2/packages_controller.rb:
```diff
@@ -2,6 +2,9 @@ module Katello
+    before_filter :find_repositories, :only => :auto_complete_name
 
     api :GET, "/packages", N_("List packages")
     api :GET, "/repositories/:repository_id/packages", N_("List packages")
```
I know we have discussed balancing package vs RPM as part of this change. Should we put an eye to the future and introduce `/rpms` based APIs? And/or start the deprecation cycle for things with `packages` ?

jlsherrill replied:
I went back and forth about this.  I feel like more than just that would need to be changed, and it might be better to go through the whole app in a separate PR.  Will file an issue if you agree

ehelms replied:
Sounds reasonable.

jlsherrill replied:
http://projects.theforeman.org/issues/11273

ehelms commented on app/controllers/katello/api/v2/packages_controller.rb:
```diff
@@ -12,10 +15,23 @@ def index
+    def auto_complete_name
+      rpms = Rpm.in_repositories(@repositories)
+      col = "#{Rpm.table_name}.name"
+      rpms = rpms.where("#{Rpm.table_name}.name ILIKE ?", "#{params[:term]}%").select(col).group(col).order(col).limit(20)
```
Where does the `20` come from?

jlsherrill replied:
It is also what scoped seach hard codes for its auto complete limits.  I'm not sure if page size is appropriate, nor that the user needs to change it.  I guess i can make it a constant somewhere.

ehelms replied:
Maybe make it a constant on the autocomplete module (https://github.com/Katello/katello/blob/fc47f03eaceef0975598efdc542acf6976724426/app/controllers/katello/concerns/filtered_auto_complete_search.rb) since that typically gets included everywhere and could connect the number to autocomplete?

ehelms commented on app/controllers/katello/api/v2/repositories_controller.rb:
```diff
@@ -39,6 +40,7 @@ class Api::V2::RepositoriesController < Api::V2::ApiController
     param :content_view_id, :number, :desc => N_("ID of a content view to show repositories in")
     param :content_view_version_id, :number, :desc => N_("ID of a content view version to show repositories in")
     param :erratum_id, String, :desc => N_("UUID of an erratum to find repositories that contain the erratum")
+    param :rpm_id, String, :desc => N_("UUID of a package to find repositories that contain the rpm")
```
Should this be `UUID of a RPM` ?

ehelms replied:
Also, are we only supporting UUID in this case?

ehelms commented on app/controllers/katello/api/v2/repositories_controller.rb:
```diff
@@ -171,10 +174,11 @@ def sync_complete
     desc "Remove content from a repository"
     param :id, :identifier, :required => true, :desc => "repository ID"
-    param 'uuids', Array, :required => true, :desc => "Array of content uuids to remove"
+    param 'uuids', Array, :required => true, :deprecated => true, :desc => "Array of content uuids to remove"
```
s/uuids/UUIDs

ehelms commented on app/controllers/katello/api/v2/repositories_controller.rb:
```diff
@@ -171,10 +174,11 @@ def sync_complete
     param :id, :identifier, :required => true, :desc => "repository ID"
-    param 'uuids', Array, :required => true, :desc => "Array of content uuids to remove"
+    param 'uuids', Array, :required => true, :deprecated => true, :desc => "Array of content uuids to remove"
+    param 'ids', Array, :required => true, :desc => "Array of content ids to remove"
```
s/ids/IDs/ ?

jlsherrill replied:
i will switch to lower case.  Seem silly to uppercase 'id.

ehelms commented on app/controllers/katello/api/v2/repositories_controller.rb:
```diff
@@ -171,10 +174,11 @@ def sync_complete
     def remove_content
-      fail _("No package uuids provided") if params[:uuids].blank?
-      respond_for_async :resource => sync_task(::Actions::Katello::Repository::RemoveContent, @repository, params[:uuids])
+      fail _("No content ids provided") if (params[:uuids] || params[:ids]).blank?
```
Could just check for `@content` to be blank?

ehelms commented on app/controllers/katello/concerns/api/v2/repository_db_content_controller.rb:
```diff
@@ -11,7 +11,15 @@ module Api::V2::RepositoryDbContentController
+          sort_options = default_sort
+        else
+          options[:custom_sort] =  default_sort
+        end
```
This is not the most obvious logic as to what is happening and why.

ehelms commented on app/models/katello/glue/pulp/rpm.rb:
```diff
@@ -0,0 +1,38 @@
+        pulp_facts['requires'].map { |entry| Katello::Util::Package.format_requires(entry) }
+      else
+        []
+      end
```
Could write this and the subsequent functions in this form:

```
def requires
  return [] unless pulp_facts['repos']
  pulp_facts['requires'].map { |entry| Katello::Util::Package.format_requires(entry) }
end
```

jlsherrill replied:
Hard to explain but i was avoiding an explicit return.  As i would count your re-write as an in-function return, but mine as a single return since its all within a single if statement.

ehelms replied:
What's the difference or reason you'd try and avoid an explicit return in this case?

ehelms replied:
Was still curious your answer to my reply above - this got eaten by the outdated diff monster.

jlsherrill replied:
We generally try to avoid things like:
```
def foo
  if a
    return 'foo'
  end
  blah.each do |item|
     do_something else
   end
   return 3 if needed
   do more stuff
   return 4
end
```

and tend to prefer a single return statement.

ehelms replied:
I am still not seeing the 'why' I guess? And in this case there is one explicit return statement in the guard clause?

jlsherrill replied:
So there are few things here.

1.  Multiple returns within a function being discouraged

Are you confused by this?  Do you disagree with this?  I swear you have
commented on my code telling me to only return once.

2.  Does my version qualify as having multiple returns

I don't think it does qualify.  The difference is subtle.  The reason I say
this is that the if statement results in a single value.  You could rewrite
it to be more obvious why it doesn't quality:

```
   to_ret = if true
                    1
                else
                     2
                end
   to_ret
```

3.  Does your version qualify as having multiple returns.

 Yes, there is an explicit version and a non-explicit version


On Mon, Aug 3, 2015 at 6:09 PM, Eric D Helms <notifications@github.com>
wrote:

> In app/models/katello/glue/pulp/rpm.rb
> <https://github.com/Katello/katello/pull/5326#discussion_r36137657>:
>
> > +    include LazyAccessor
> > +
> > +    included do
> > +      lazy_accessor :pulp_facts, :initializer => :backend_data
> > +
> > +      lazy_accessor :description, :license, :buildhost, :vendor, :relativepath, :children, :checksumtype,
> > +                    :changelog, :group, :size, :url, :build_time, :group,
> > +                    :initializer => :pulp_facts
> > +    end
> > +
> > +    def requires
> > +      if pulp_facts['requires']
> > +        pulp_facts['requires'].map { |entry| Katello::Util::Package.format_requires(entry) }
> > +      else
> > +        []
> > +      end
>
> I am still not seeing the 'why' I guess? And in this case there is one
> explicit return statement in the guard clause?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Katello/katello/pull/5326/files#r36137657>.
>

ehelms replied:
Likely the multiple returns comes more from JavaScript where, due to the language, it is typically encouraged to have only one return per function and use variable assignment instead. With Ruby's syntax structure I think, as of late, using guard clauses makes smaller, nice to read code that checks for the nil/empty/simply case and quickly handles it with the complex case following after (http://c2.com/cgi/wiki?GuardClause, https://www.youtube.com/watch?v=8bZh5LMaSmE). I am by no means requiring a change in this manner, more just using this as an opportunity to explore the thought between the two ways of tackling the design while I iterate test and review on the PR.

jlsherrill replied:
I can see arguments both ways, and i'm really not sure why i debated this
so long, but in this particular case i just tend to prefer the if/else.  If
you had written the code the way you proposed I probably wouldn't have
given it a second thought.  If it were a longer function and it made sense
in context, a 'guard' clause may have felt more natural.

On Mon, Aug 3, 2015 at 7:26 PM, Eric D Helms <notifications@github.com>
wrote:

> In app/models/katello/glue/pulp/rpm.rb
> <https://github.com/Katello/katello/pull/5326#discussion_r36143224>:
>
> > +    include LazyAccessor
> > +
> > +    included do
> > +      lazy_accessor :pulp_facts, :initializer => :backend_data
> > +
> > +      lazy_accessor :description, :license, :buildhost, :vendor, :relativepath, :children, :checksumtype,
> > +                    :changelog, :group, :size, :url, :build_time, :group,
> > +                    :initializer => :pulp_facts
> > +    end
> > +
> > +    def requires
> > +      if pulp_facts['requires']
> > +        pulp_facts['requires'].map { |entry| Katello::Util::Package.format_requires(entry) }
> > +      else
> > +        []
> > +      end
>
> Likely the multiple returns comes more from JavaScript where, due to the
> language, it is typically encouraged to have only one return per function
> and use variable assignment instead. With Ruby's syntax structure I think,
> as of late, using guard clauses makes smaller, nice to read code that
> checks for the nil/empty/simply case and quickly handles it with the
> complex case following after (http://c2.com/cgi/wiki?GuardClause,
> https://www.youtube.com/watch?v=8bZh5LMaSmE). I am by no means requiring
> a change in this manner, more just using this as an opportunity to explore
> the thought between the two ways of tackling the design while I iterate
> test and review on the PR.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Katello/katello/pull/5326/files#r36143224>.
>

ehelms commented on app/models/katello/repository.rb:
```diff
@@ -298,8 +301,11 @@ def clone_id(env, content_view, version = nil)
+        self.rpms.where("#{Rpm.table_name}.filename NOT in (?)", errata_filenames)
+      else
+        self.rpms
+      end
```
Could do:

```
def packages_without_errata
  return self.rpms if errata_filenames.empty?
  self.rpms.where("#{Rpm.table_name}.filename NOT in (?)", errata_filenames)
end
```

ehelms commented on app/models/katello/repository.rb:
```diff
@@ -483,8 +468,37 @@ def import_system_applicability
+      elsif docker?
+        self.docker_images
+      else
+        fail "Data type not supported"
```
This isn't the most obvious of failure messages if you were elsewhere in the stack (or it somehow bubbled up to the user)

jlsherrill replied:
It really should never bubble up to the user, and I Image once you add puppet you'll remove it

jlsherrill replied:
but i will update anyways

ehelms commented on app/models/katello/repository.rb:
```diff
@@ -524,5 +538,15 @@ def ensure_docker_repo_unprotected
+
+      # destroy any orphan docker images
+      images.reload.each do |image|
+        image.destroy if image.repositories.count < 1
```
Might could just check `image.repositories.empty?`

jlsherrill replied:
was a copy paste

ehelms commented on app/models/katello/rpm.rb:
```diff
@@ -0,0 +1,65 @@
+module Katello
+  class Rpm < Katello::Model
+    include Glue::Pulp::PulpContentUnit
+    include Glue::Pulp::Rpm
```
Are we forgoing the `if Katello.config.use_pulp` check?

jlsherrill replied:
I think we should.  we don't need them in tests.  Any reason to keep them?

ehelms replied:
I suppose convention. If these types of checks and guard are no longer needed we ought to just remove them wholesale.

jlsherrill replied:
I think that ship has sailed, but i can open an issue to remove the rest

ehelms commented on app/models/katello/rpm.rb:
```diff
@@ -0,0 +1,65 @@
+    scoped_search :on => :sourcerpm, :complete_value => true
+    scoped_search :on => :checksum
+
+    before_save lambda { |rpm| rpm.summary= rpm.summary.truncate(255) unless rpm.summary.blank? }
```
Should this be `rpm.summary =` (note the missing blank) ? Might explain why I was seeing no summaries?

jlsherrill replied:
the cause was actually the PULP_INDEX_FIELDS did not include summary.  Will change so this list is shared.

ehelms commented on engines/bastion_katello/app/assets/javascripts/bastion_katello/packages/details/package-details-repositories.controller.js:
```diff
@@ -0,0 +1,57 @@
+ */
+angular.module('Bastion.packages').controller('PackageDetailsRepositoriesController',
+['$scope', '$q', 'Nutupane', 'Repository', 'Environment', 'ContentView', 'CurrentOrganization',
+function ($scope, $q, Nutupane, Repository, Environment, ContentView, CurrentOrganization) {
```
Indentation is off

ehelms commented on engines/bastion_katello/app/assets/javascripts/bastion_katello/packages/details/package-details-repositories.controller.js:
```diff
@@ -0,0 +1,57 @@
+angular.module('Bastion.packages').controller('PackageDetailsRepositoriesController',
+['$scope', '$q', 'Nutupane', 'Repository', 'Environment', 'ContentView', 'CurrentOrganization',
+function ($scope, $q, Nutupane, Repository, Environment, ContentView, CurrentOrganization) {
+    var repositoriesNutupane, environment, contentView, params = {
```
Mind doing params on a separate line since it is a non-blank object?

jlsherrill commented:
On Fri, Jul 31, 2015 at 3:42 PM, Eric D Helms <notifications@github.com>
wrote:

> 1) Strangely it isn't suggesting x86_64 for autocomplete when doing arch.
>
 Caused by: https://github.com/wvanbergen/scoped_search/pull/134

> 2) I get no autocomplete suggestions for checksum
>
   I turned this off on purpose.   I don't think its useful?

> 3) *.el7 changes itself to 0.0.el7 -- can I search by 'dist' in the
> release?
>
This would do that:

release ~ %.el7

I'm not sure what you mean by 'changes itself to'.  There is a package (at
least for me that has a release of 0.0.el7, maybe that is what you are
seeing?)

> 4) Not specifically this PRs issue, but if you view a RHEL repo, and click
> the package link it takes you to the list of packages with the remove
> action available which throws a 500 when you try but never shows the user
> an error.
>
Guessing this is not a new issue?  Will file an issue if not


> 5) Should the 'Packages' menu link be 'RPMs' ?
>
I debated this, but i feel we'd want to overhaul the entire UI to be
consistent.  Most likely in a separate issue.


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Katello/katello/pull/5326#issuecomment-126793972>.
>

ehelms commented:
On Sat, Aug 1, 2015 at 4:34 PM, Justin Sherrill <notifications@github.com>
wrote:

> On Fri, Jul 31, 2015 at 3:42 PM, Eric D Helms <notifications@github.com>
> wrote:
>
> > 1) Strangely it isn't suggesting x86_64 for autocomplete when doing arch.
> >
> Caused by: https://github.com/wvanbergen/scoped_search/pull/134
>
> > 2) I get no autocomplete suggestions for checksum
> >
> I turned this off on purpose. I don't think its useful?
>

I suppose it just seems inconsistent. An autocomplete option that never
autocompletes?


>
> > 3) *.el7 changes itself to 0.0.el7 -- can I search by 'dist' in the
> > release?
> >
> This would do that:
>
> release ~ %.el7
>
> I'm not sure what you mean by 'changes itself to'. There is a package (at
> least for me that has a release of 0.0.el7, maybe that is what you are
> seeing?)
>

If I entered "release = *.el7", it would come back changed to "release =
0.0.el7".


>
> > 4) Not specifically this PRs issue, but if you view a RHEL repo, and
> click
> > the package link it takes you to the list of packages with the remove
> > action available which throws a 500 when you try but never shows the user
> > an error.
> >
> Guessing this is not a new issue? Will file an issue if not
>
>
> > 5) Should the 'Packages' menu link be 'RPMs' ?
> >
> I debated this, but i feel we'd want to overhaul the entire UI to be
> consistent. Most likely in a separate issue.
>
>
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/Katello/katello/pull/5326#issuecomment-126793972>.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Katello/katello/pull/5326#issuecomment-126952139>.
>



-- 
Eric D. Helms
Red Hat Engineering
Ph.D. Student - North Carolina State University

jlsherrill commented:
On Sun, Aug 2, 2015 at 9:49 AM, Eric D Helms <notifications@github.com>
wrote:

> On Sat, Aug 1, 2015 at 4:34 PM, Justin Sherrill <notifications@github.com>
> wrote:
>
> > On Fri, Jul 31, 2015 at 3:42 PM, Eric D Helms <notifications@github.com>
> > wrote:
> >
> > > 1) Strangely it isn't suggesting x86_64 for autocomplete when doing
> arch.
> > >
> > Caused by: https://github.com/wvanbergen/scoped_search/pull/134
> >
> > > 2) I get no autocomplete suggestions for checksum
> > >
> > I turned this off on purpose. I don't think its useful?
> >
>
> I suppose it just seems inconsistent. An autocomplete option that never
> autocompletes?
>

If it wasn't there at all the user wouldn't know they could search by it.
I can turn it on if you feel strongly about it.


>
>
> >
> > > 3) *.el7 changes itself to 0.0.el7 -- can I search by 'dist' in the
> > > release?
> > >
> > This would do that:
> >
> > release ~ %.el7
> >
> > I'm not sure what you mean by 'changes itself to'. There is a package (at
> > least for me that has a release of 0.0.el7, maybe that is what you are
> > seeing?)
> >
>
> If I entered "release = *.el7", it would come back changed to "release =
> 0.0.el7".
>

I think what you are seeing is a 'quirk' in the auto-complete widget.
tying a search and hitting enter will search on the first item in the list.


>
>
> >
> > > 4) Not specifically this PRs issue, but if you view a RHEL repo, and
> > click
> > > the package link it takes you to the list of packages with the remove
> > > action available which throws a 500 when you try but never shows the
> user
> > > an error.
> > >
> > Guessing this is not a new issue? Will file an issue if not
> >
> >
> > > 5) Should the 'Packages' menu link be 'RPMs' ?
> > >
> > I debated this, but i feel we'd want to overhaul the entire UI to be
> > consistent. Most likely in a separate issue.
> >
> >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <https://github.com/Katello/katello/pull/5326#issuecomment-126793972>.
> >
> > >
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/Katello/katello/pull/5326#issuecomment-126952139>.
> >
>
>
>
> --
> Eric D. Helms
> Red Hat Engineering
> Ph.D. Student - North Carolina State University
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Katello/katello/pull/5326#issuecomment-127025430>.
>

jlsherrill commented:
6) Select checkboxes exist next to RPMs on the 'Packages' page but no actions are present
fixed
7) No summaries are present for any of the RPMs ?
fixed
8) No packages seem to show anything under 'Requires' when browsing the dependencies tab.
fixed (angular doesn't like to sort with duplicates)
9) When viewing an errata, it lists the affected packages, what about turning those into links to the package details pages for each (could be a future issue) ?
done

jlsherrill commented:
[test]

ehelms commented on engines/bastion_katello/app/assets/javascripts/bastion_katello/packages/details/package-details-repositories.controller.js:
```diff
@@ -0,0 +1,59 @@
+        $scope.detailsTable = repositoriesNutupane.table;
+        $scope.detailsTable.initialLoad = false;
+        repositoriesNutupane.masterOnly = true;
+        repositoriesNutupane.searchKey = 'repositoriesSearch';
```
This seems to not be working, I wonder if it broke at some point. The search from the main table is bleeding into the details table.

jlsherrill replied:
filed http://projects.theforeman.org/issues/11289

jlsherrill commented:
[test]

ehelms commented:
@jlsherrill did this require a lot of updates after package groups was merged or minor?

jlsherrill commented:
@ehelms very minor, mostly just one-line conflicts.

The biggest difference is https://github.com/Katello/katello/pull/5326/files#diff-c1185b4820e5f9eecbefe1d60381b662L86

as I was able to unify some of that logic to be reused elsewhere.

ehelms commented:
ACK

jlsherrill commented:
I realized that uploading an rpm does not work properly, need to fix that!

johnpmitsch commented on app/controllers/katello/concerns/api/v2/repository_db_content_controller.rb:
```diff
@@ -83,12 +93,7 @@ def filter_by_environment(environment, collection)
-          @resource = resource_class.where(:uuid => params[:id]).first
-        end
+        @resource = resource_class.with_identifiers(params[:id]).first
 
```
Does this change still allow querying by uuid in the API?

jlsherrill replied:
it does!  I need to use it in multiple places so I created a helper method

johnpmitsch commented:
Maybe I am missing something, but is there a way to change pages in the packages tab of the UI? I see 'Showing 20 of 3819 (3819 Total)', but can't get to the next 20

johnpmitsch commented:
Searching by filename in the search bar, and clicking on a package that is returned as a result, then going to the repositories tab, I get this traceback http://fpaste.org/254453/43940358/ probably as a result of two search boxes on the page being filled in with a URL parameter?

cfouant commented on engines/bastion_katello/app/assets/javascripts/bastion_katello/packages/views/packages.html:
```diff
@@ -0,0 +1,19 @@
+<span page-title>{{ 'Errata' | translate }}</span>
```
This should be 'Packages' rather than 'Errata'

ehelms commented:
You are seeing a result of the bug with the two search boxes overlapping
each other that Justin filed.

On Wed, Aug 12, 2015 at 2:22 PM, John Mitsch <notifications@github.com>
wrote:

> Searching by filename in the search bar, and clicking on a package that is
> returned as a result, then going to the repositories tab, I get this
> traceback http://fpaste.org/254453/43940358/ probably as a result of two
> search boxes on the page being filled in with a URL parameter?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Katello/katello/pull/5326#issuecomment-130400572>.
>



-- 
Eric D. Helms
Red Hat Engineering
Ph.D. Student - North Carolina State University

ehelms commented:
This sounds like infinite scroll acting up on larger browser heights --
guessing you see a gap between the last package and the bottom of the
browser? If you shrink your browser height down and scroll down do you find
that it loads?

On Wed, Aug 12, 2015 at 2:15 PM, John Mitsch <notifications@github.com>
wrote:

> Maybe I am missing something, but is there a way to change pages in the
> packages tab of the UI? I see 'Showing 20 of 3819 (3819 Total)', but can't
> get to the next 20
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Katello/katello/pull/5326#issuecomment-130398765>.
>



-- 
Eric D. Helms
Red Hat Engineering
Ph.D. Student - North Carolina State University

johnpmitsch commented:
@ehelms I see, didn't know there was a bug filed about the two search boxes. Just tested out the infinite scroll, it is the browser height. Is this a know issue or perhaps its a common convention? IMO, It seems like it could be misleading to users.

ehelms commented:
Here is the issue @jlsherrill filed for reference: http://projects.theforeman.org/issues/11289

The infinite scroll issue might be a result of @waldenraines switching Bastion's infinite scroll library as this sounds like a regression. @johnpmitsch please file a bug (http://projects.theforeman.org/projects/bastion/issues/new) so that we can look in to it.

johnpmitsch commented:
@ehelms thanks, filed it here http://projects.theforeman.org/issues/11343

jlsherrill commented:
Rpm uploads fixed and cfoo's title chnage fixed.

cfouant commented:
@jlsherrill ACK from me

ehelms commented:
ACK
